foundry.toml

```diff
@@ -1,9 +1,32 @@
 [profile.default]
-src = 'src'
-out = 'out'
-libs = ['lib']
+  optimizer = true
+  optimizer_runs = 10_000_000
+  solc_version = "0.8.16"
```

Comment:
Does this build for you? I get errors because the Aave contracts are pinned to 0.8.10

```
error[5333]: ParserError: Source file requires different compiler version (current compiler is 0.8.16+commit.07a7930e.Darwin.appleclang) - note that nightly builds are considered to be strictly less than the released version
 --> lib/aave-v3-core/contracts/protocol/tokenization/base/ScaledBalanceTokenBase.sol:2:1:
  |
2 | pragma solidity 0.8.10;
```

Reply:
Haha, no. My bad

Reply:
I'm thinking we should just leave out the `solc_version` config for this repo, given that it contains source code that we don't control (and thus solc versions that we don't control). It seems better to just let foundry auto-detect solc version on a case-by-case basis.

Reply:
Done in bab4ade

Reply:
I'm fine with that, though I do prefer an explicit version (just easier to audit / know what solc version you are using). Feel free to resolve this comment yourself.

Reply:
Oh, we actually cannot pin a version here, because some files (Aave) require 0.8.10, and others (`FractionaPool.sol` and `FractionaPool.t.sol`) require 0.8.13. So the solutions are:
- Don't pin a version: some files get compiled with 0.8.10, and others with 0.8.17
- Pin 0.8.10 and change our contracts to use 0.8.10

We probably should use 0.8.17 whenever we can, so feel free to resolve this comment yourself

Reply:
Pushed f015a53 to clarify this, feel free to tweak phrasing